README.md

```diff
@@ -27,13 +27,25 @@ overhead.
 
 :star2: For more detailed technical discussion, please check our [Wiki][wiki], [Engineering Blog][blog], and paper (WIP).
 
-## Installation
-
-NilAway is implemented using the standard [go/analysis](https://pkg.go.dev/golang.org/x/tools/go/analysis) framework, 
-making it easy to integrate with existing analyzer drivers (e.g., [golangci-lint](https://github.com/golangci/golangci-lint),
-[nogo](https://github.com/bazelbuild/rules_go/blob/master/go/nogo.rst), or 
-[running as a standalone checker](https://pkg.go.dev/golang.org/x/tools/go/analysis/singlechecker)). Here, we list the 
-instructions for running NilAway as a standalone checker. More integration supports will be added soon.
+## Running NilAway
+
+NilAway is implemented using the standard [go/analysis][go-analysis], making it easy to integrate with existing analyzer
+drivers (i.e., [golangci-lint][golangci-lint], [nogo][nogo], or [running as a standalone checker][singlechecker]).
+
+> [!IMPORTANT]  
+> Due to the sophistication of the analyses that NilAway does, it tries to cache its findings about a particular 
+> package via the [Fact Mechanism][fact-mechanism] from the [go/analysis][go-analysis] framework. Therefore, it is 
+> _highly_ recommended to leverage a driver that supports modular analysis (i.e., bazel/nogo or golangci-lint, but _not_
+> the standalone checker since it stores all facts in memory) for better performance on large projects.
+
+> [!IMPORTANT]  
+> By default, NilAway analyzes _all_ Go code, including the standard libraries and dependencies. This helps NilAway 
+> better understand the code form dependencies and reduce its false negatives. However, this would also incur a 
+> significant performance cost (only once for drivers with modular support) and increase the number of non-actionable 
+> errors in dependencies, for large Go projects with a lot of dependencies.
+> 
+> You can use the [`include-pkgs`][include-pkgs-flag] flag to only include analysis of first-party code, where NilAway
+> will apply optimistic defaults for the out-of-scope Go code (meaning a potential increase of false negatives).
```

Comment:
I think we can perhaps rephrase the text to keep it more user oriented. Suggestion:
"""
We highly recommend using the [`include-pkgs`][include-pkgs-flag] flag to narrow down the analysis to your project's code exclusively. This directs NilAway to skip analyzing dependencies (e.g., third-party libraries), allowing you to focus solely on potential nil panics reported by NilAway in your first-party code!
"""

Reply:
Great point! Replaced this part with your suggestions :)